src/usr/local/www/services_dhcp_edit.php

```diff
@@ -385,8 +385,19 @@ function staticmaps_sort($ifgui) {
 
 // Get our MAC address
 $ip = $_SERVER['REMOTE_ADDR'];
-$mymac = `/usr/sbin/arp -an | grep '('{$ip}')' | cut -d" " -f4`;
-$mymac = str_replace("\n", "", $mymac);
+
+switch (is_ipaddr($ip)) {
+	case 4:
+		$mymac = exec("/usr/sbin/arp -an | grep '('{$ip}')' | head -n 1 | cut -d' ' -f4");
+		$mymac = str_replace("\n", "", $mymac);
+        break;
+    case 6:
+		$mymac = exec("/usr/sbin/ndp -na | /usr/bin/grep '{$ip}' | /usr/bin/head -n 1 | /usr/bin/awk '{ print $2 }'");
+		$mymac = str_replace("\n", "", $mymac);
+		break;
+	default:
+		unset($mymac);
+}
```

Comment:
Ouch, I was blink when I looked around. Yes, but it needs to be changed to work with IPv6 at least